modules/op-using-a-custom-database-in-tekton-hub.adoc

```diff
@@ -0,0 +1,74 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-tekton-hub-with-openshift-pipelines.adoc
+
+:_content-type: PROCEDURE
+[id="using-a-custom-database-in-tekton-hub_{context}"]
+= Optional: Using a custom database in {tekton-hub}
+
+[role="_abstract"]
+After installation of {tekton-hub} on a cluster, administrators can use a custom database instead of the default PostgreSQL database installed by the Operator. You can use your custom database with `db-migration`, `api`, and `ui` interfaces provided by {tekton-hub}.
+
+[discrete]
+.Procedure
+
+. Create a secret named `tekton-hub-db` in the target namespace with the following keys: 
+* `POSTGRES_HOST`
+* `POSTGRES_DB`
+* `POSTGRES_USER`
+* `POSTGRES_PASSWORD`
+* `POSTGRES_PORT`
++
+.Example: Custom database secrets
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: tekton-hub-db
+  labels:
+    app: tekton-hub-db
+type: Opaque
+stringData:
+  POSTGRES_HOST: <The name of the host of the database>
+  POSTGRES_DB: <Name of the database>
+  POSTGRES_USER: <The name of user account>
+  POSTGRES_PASSWORD: <The password of user account>
+  POSTGRES_PORT: <The port that the database is listening on>
+...
+----
++
+[NOTE]
+====
+The default target namespace is `openshift-pipelines`.
+====
+
+. In the `TektonHub` CR, set the value of the database secret attribute to `tekton-hub-db`.
++
+.Example: Adding custom database secret 
+[source,yaml]
+----
+apiVersion: operator.tekton.dev/v1alpha1
+kind: TektonHub
+metadata:
+  name: hub
+spec:
+  targetNamespace: openshift-pipelines
+  db:                      
+    secret: tekton-hub-db 
+  api:
+    hubConfigUrl: https://raw.githubusercontent.com/tektoncd/hub/main/config.yaml
+    catalogRefreshInterval: 30m
+...
+----
+
+. Apply the updated `TektonHub` CR.
++
+[source,terminal]
+----
+$ oc apply -f <tekton-hub-cr>.yaml
+----
+
+. Migrate data from the default database to the custom database.
+
+After successful migration, the Operator automatically deletes the default database.
```

Comment:
So not after successful migration, once the secret is created by the user and when the cr is applied we delete the default database, so may be we can change it to 

Once the CR is applied with the user's database secret and if user has already installed Tekton Hub with default database before, then in that case default database will be deleted and other Tekton Hub services will be connected to user's custom database